airflow/contrib/operators/databricks_operator.py

```diff
@@ -0,0 +1,196 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+import time
+
+from airflow.exceptions import AirflowException
+from airflow.contrib.hooks.databricks_hook import DatabricksHook
+from airflow.models import BaseOperator
+
+LINE_BREAK = ("-" * 80)
+
+
+class DatabricksSubmitRunOperator(BaseOperator):
+    """
+    Submits an ephemeral run to Databricks.
+
+    https://docs.databricks.com/api/latest/jobs.html#runs-submit
+
+    There are two ways to instantiate this operator.
+
+    In the first way, you can take the JSON payload that you typically use
+    to call the ``api/2.0/jobs/runs/submit`` endpoint and pass it directly
+    to our ``DatabricksSubmitRunOperator`` through the ``json`` parameter.
+    For example ::
+        params = {
+          "new_cluster": {
+            "spark_version": "2.1.0-db3-scala2.11",
+            "num_workers": 2
+          },
+          "notebook_task": {
+            "notebook_path": "/test",
+          },
+        }
+        DatabricksSubmitRunOperator(task_id='notebook_run', json=params)
+
+    Another way to accomplish the same thing is to use the named parameters
+    of the ``DatabricksSubmitRunOperator`` directly. Note that there is exactly
+    one named parameter for each top level parameter in the ``runs/submit``
+    endpoint. In this method, your code would look like this: ::
+        new_cluster = {
+          "spark_version": "2.1.0-db3-scala2.11",
+          "num_workers": 2
+        }
+        notebook_task: {
+          "notebook_path": "/test",
+        }
+        DatabricksSubmitRunOperator(
+            task_id='notebook_run',
+            new_cluster=new_cluster,
+            notebook_task=notebook_task)
+
+    In the case where both the json parameter is provided **AND** the named parameters
+    are provided, they will be merged together. If there are conflicts during the merge,
+    the named parameters will take precedence and override the top level ``json`` keys.
+
+    :param json: A JSON object containing API parameters which will be passed
```

Comment:
A fair point. I was thinking about data, but if your API expects json, it might be fine. SGTM